pkg/controller/operators/olm/operator.go

```diff
@@ -827,6 +828,42 @@ func (a *Operator) syncNamespace(obj interface{}) error {
 		return err
 	}
 
+	// Filter the operator associated with this namespace
+	groups := make([]*v1.OperatorGroup, 0)
+	for _, og := range operatorGroupList {
+		if og.GetNamespace() == namespace.GetName() {
+			groups = append(groups, og)
+		}
+	}
```

Comment:
Non-Blocking Nit: Given that the lister retrieves the `OperatorGroups` from a cache it might be cleaner to filter by namespace above. An additional call to the cache can be made below for the bit that requires all `OperatorGroups`.